### DISCUSSION TITLE
How can I define default DateTime in 2.x style?

### DISCUSSION THREAD

**Discussion by davuses** [sqlalchemy/sqlalchemy#10189] [Usage Questions]
This is how I define a default DateTime in 1.x style:
```py
update_time = Column(
    DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
```
The database is MySQL. How can I do this in 2.x style?

**Comment by zzzeek** [comment]
you can substitute `mapped_column()` directly for `Column()` with no change, is the most straightforward:

```py
update_time = mapped_column(
    DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
```

you can add annotations but they should be optional here

```py
update_time: Mapped[datetime] = mapped_column(
    DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
```

there's then all the usual ways to automate generating a column like this, such as using `Annotated`

```py
import datetime

from typing_extensions import Annotated

from sqlalchemy import func
from sqlalchemy.orm import mapped_column


timestamp = Annotated[
    datetime.datetime,
    mapped_column(
      DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
]

class MyClass(...):
    # ...

    update_time: Mapped[timestamp]

```

**Reply by davuses** [reply]
```py
update_time: Mapped[datetime] = mapped_column(
    DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
)
```
Pylint throws this error to me: `func.now is not callable Pylint[E1102:not-callable]`

**Reply by CaselIT** [reply]
That seems a bug of pylint.
Now is defined here https://github.com/sqlalchemy/sqlalchemy/blob/1377f33075a0e4230c88506d2bfd1e847a336da5/lib/sqlalchemy/sql/functions.py#L1008

**Comment by Kyle-sandeman-mrdfood** [comment]
It seems like you can use `now` from `sqlalchemy.sql.functions` instead. PyLint passes for me and I've confirmed it emits the same DDL for SQLite/Postgresql
